crypto/provider_core.c

```diff
@@ -1118,7 +1118,6 @@ static OPENSSL_CORE_CTX *core_get_libctx(const OSSL_CORE_HANDLE *handle)
      * that does not apply here. Here |prov| == NULL can happen only in
      * case of a coding error.
      */
-    (void)ossl_assert(prov != NULL);
```

Comment:
In the case where we don't want to handle the error in a non-debug build we can use standard `assert`.

Reply:
OK, I'll use that.